genconfig/provider_map.yaml

```diff
@@ -45,3 +45,7221 @@ akamai:
+      - lankadeepa.lk
+      - nflstreams.to
+      - cinecalidad3.io
+      - ticketmaster.co.uk
+      - lol.ps
+      - xxjsq2.com
+      - north-plus.net
+      - dafiti.com.br
+      - treasury.gov
+      - greencdn.io
+      - sympla.com.br
+      - 1001freefonts.com
+      - campuslabs.com
+      - orange.es
+      - ionos.co.uk
+      - upbit.com
+      - torrentjogos.net
+      - itsfoss.com
+      - portal.restaurant
+      - 4download.net
+      - sbi.co.in
+      - tjx.com
+      - kurzy.cz
+      - bcv.org.ve
+      - blogdepelis.io
+      - bflix.to
+      - momplaybook.com
+      - shaheed4u.motorcycles
+      - d-id.com
+      - screenconnect.com
+      - lizhi.io
+      - 1inch.io
+      - azerisport.com
+      - airbnb.mx
+      - lendingtree.com
+      - bobbyhadz.com
+      - cricut.com
+      - cognitoforms.com
+      - centennialcollege.ca
+      - yopmail.com
+      - krloli.com
+      - smiles.com.br
+      - channelstv.com
+      - mobafire.com
+      - google.tn
+      - saoniuhuo.com
+      - ig.com
+      - groups.io
+      - bodybuilding.com
+      - f2m.top
+      - sezonlukdizi3.com
+      - vinmec.com
+      - sumup.com
+      - iaai.com
+      - sportsdirect.com
+      - wyndhamhotels.com
+      - glavkniga.ru
+      - denik.cz
+      - qastack.cn
+      - topgmovies.xyz
+      - doramasqueen.com
+      - startech.com.bd
+      - camsstream.com
+      - shotcut.org
+      - apkdone.com
+      - rossko.ru
+      - ygoprodeck.com
+      - uplabs.com
+      - bulma.io
+      - 123-movies.sb
+      - epark.jp
+      - kahoot.com
+      - memo.wiki
+      - madisoft.it
+      - google.com.gt
+      - whattomine.com
+      - teachtci.com
+      - zetfix-online.net
+      - handelsblatt.com
+      - sega.jp
+      - hessen.de
+      - lorefree.com
+      - z2.fm
+      - gimkit.com
+      - codehs.com
+      - hirunews.lk
+      - restaurantguru.com
+      - articlesknight.com
+      - xidol.net
+      - zocdoc.com
+      - cat.com
+      - metruyencv.com
+      - sherdog.com
+      - collegenet.com
+      - truenas.com
+      - sputniknews.com
+      - funcionpublica.gov.co
+      - pabbly.com
+      - cvshealth.com
+      - openvino.ai
+      - nperf.com
+      - faharas.net
+      - lastsecond.ir
+      - 404dh.icu
+      - payeer.com
+      - freesound.org
+      - wildberries.by
+      - myhermes.de
+      - news.ru
+      - unionbankofindia.co.in
+      - detmir.ru
+      - kitset.ir
+      - banimode.com
+      - animension.to
+      - instamojo.com
+      - shuge.org
+      - free-mp3-download.net
+      - ccma.cat
+      - tnt.com
+      - hse.ru
+      - siberianhealth.com
+      - wink.ru
+      - delhigovt.nic.in
+      - erase.bg
+      - ped-kopilka.ru
+      - freefrontend.com
+      - studopedia.ru
+      - yurticikargo.com
+      - ketabrah.ir
+      - vesti.bg
+      - parscoders.com
+    ir:
+      usearbitrarysnis: true
+      arbitrarysnis:
+    cn:
+      usearbitrarysnis: true
```

Comment:
(I know I already approved this in the last PR, but I've given it more thought)

Domain fronting is reliable in China, so let's not use arbitrary SNIs there for the moment.  Let's keep the China-specific SNI list here for future use though.  Let's use arbitrary SNIs in Iran, Myanmar, Russia, and the UAE.  I'm not sure to be honest whether it will help in all of those countries, but it'll be interesting to see!

Here's some data (from Honeycomb) on domain fronting error rates over the last week.  I'm basing the decisions above on this:

![image](https://github.com/user-attachments/assets/db8355da-093b-409a-998d-ff2081f43017)

And here's the query if you want to play around with it yourself:

<details>
  <summary>Query</summary>

![image](https://github.com/user-attachments/assets/1b31fb06-148a-4764-bb63-50f934c33393)

</details>